ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -794,6 +794,19 @@ impl<'a> Assemble<'a> {
                             }
                             _ => todo!(),
                         },
+                        BinOp::Shl => match bit_size {
+                            0 => unreachable!(),
+                            32 => dynasm!(self.asm; shl Rd(lhs_reg.code()), v),
+                            1..=64 => {
+                                // Ensure we shift in zeros at the most-significant bits.
```

Comment:
Unless I'm mistaken, I don't think you need any sign or zero extend stuff here. `shl` will always shift in zeros at the LSB and the undefined bits towards the MSB are irrelevant, right?

Reply:
Oops, you're right -- I wasn't thinking! Fixed in ec22a96.